# eslint-plugin-warp-drive

| Rule | 🏷️ | ✨ |
| ---- | -- | -- |
| `no-invalid-resource-types` | 🏆 | ✅🛠️ |

## Rule Details

### Notes

## Configuration

## Examples

Examples of **incorrect** code for this rule:


Examples of **correct** code for this rule:



## References

- 
