# eslint-plugin-warp-drive

| Rule | ğŸ·ï¸ | âœ¨ |
| ---- | -- | -- |
| `no-invalid-resource-ids` | ğŸ† | âœ…ğŸ› ï¸ |

## Rule Details

### Notes

## Configuration

## Examples

Examples of **incorrect** code for this rule:


Examples of **correct** code for this rule:



## References

- 
