diff --git a/addon/components/vertical-collection/template.hbs b/addon/components/vertical-collection.hbs
similarity index 100%
rename from addon/components/vertical-collection/template.hbs
rename to addon/components/vertical-collection.hbs
diff --git a/addon/components/vertical-collection/component.js b/addon/components/vertical-collection.js
similarity index 99%
rename from addon/components/vertical-collection/component.js
rename to addon/components/vertical-collection.js
index f67198cabddc69caa2e13018afbf9524842c8808..3bc8f55dfe97c7adfac9d6d14583d32b580a2c79 100644
--- a/addon/components/vertical-collection/component.js
+++ b/addon/components/vertical-collection.js
@@ -6,8 +6,7 @@ import { empty, readOnly } from '@ember/object/computed';
 import Component from '@ember/component';
 import { get, computed } from '@ember/object';
 import { run } from '@ember/runloop';
-import layout from './template';
-import { ViewportContainer } from '../../-private';
+import { ViewportContainer } from '../-private/index';
 
 import { scheduler, Token } from 'ember-raf-scheduler';
 
@@ -16,7 +15,7 @@ import {
   DynamicRadar,
   StaticRadar,
   objectAt
-} from '../../-private';
+} from '../-private/index';
 
 /*
  * BEGIN DEBUG HELPERS
@@ -179,7 +178,6 @@ class Visualization {
  */
 
 const VerticalCollection = Component.extend({
-  layout,
 
   tagName: '',
 
@@ -484,7 +482,7 @@ const VerticalCollection = Component.extend({
     }
 
     /* Public methods to Expose to parent
-      
+
       Usage:
 
       Template:
@@ -492,7 +490,7 @@ const VerticalCollection = Component.extend({
       <VerticalCollection @registerAPI={{action "registerAPI"}} />
 
       Component:
-      
+
       export default Component.extend({
         actions: {
           registerAPI(api) {
@@ -504,7 +502,7 @@ const VerticalCollection = Component.extend({
           collectionAPI.scrollToItem(index);
         }
       });
-        
+
       Need to pass this property in the vertical-collection template
       Listen in the component actions and do your custom logic
        This API will have below methods.
diff --git a/package.json b/package.json
index f75cd8125984da8bdc00fbaa7d9053078262d247..ea4dc1c72c0191280f6411bb886a34f4af75c9e2 100644
--- a/package.json
+++ b/package.json
@@ -58,7 +58,8 @@
     "ember-cli-babel": "^7.12.0",
     "ember-cli-htmlbars": "^6.0.0",
     "ember-cli-version-checker": "^5.1.2",
-    "ember-raf-scheduler": "^0.3.0"
+    "ember-raf-scheduler": "^0.3.0",
+    "ember-auto-import": "*"
   },
   "devDependencies": {
     "@ember/optional-features": "^2.0.0",
