diff --git a/qunit/qunit.js b/qunit/qunit.js
index 460e0e58227234eef7ee72876f13b7fec65198c0..3c3a786614c412e1e91b8b8d31b78cd9f3bd2b2c 100644
--- a/qunit/qunit.js
+++ b/qunit/qunit.js
@@ -8,6 +8,10 @@
  */
 (function () {
   'use strict';
+  function getKeys(obj) {
+    if (!obj) { return []; }
+      return Object.keys(obj).concat(getKeys(Object.getPrototypeOf(obj)));
+  }
 
   function _arrayLikeToArray(r, a) {
     (null == a || a > r.length) && (a = r.length);
@@ -1049,10 +1053,7 @@
             return '[object Object]';
           }
           dump.up();
-          var keys = [];
-          for (var key in map) {
-            keys.push(key);
-          }
+          var keys = getKeys(map);
 
           // Some properties are not always enumerable on Error objects.
           var nonEnumerableProperties = ['message', 'name'];
@@ -7041,6 +7042,9 @@
       appendToolbar(beginDetails);
     }
     function appendTest(name, testId, moduleName) {
+      if (window$1.DISABLE_QUNIT_HTML_REPORTER) {
+        return;
+      }
       var tests = id('qunit-tests');
       if (!tests) {
         return;
@@ -7229,6 +7233,13 @@
       assertList.appendChild(assertLi);
     });
     QUnit.testDone(function (details) {
+      // This test passed if it has no unexpected failed assertions
+      var testPassed = details.failed > 0 ? details.todo : !details.todo;
+      if (!testPassed) {
+        stats.failedTests.push(details.testId);
+      }
+      stats.completed++;
+
       var tests = id('qunit-tests');
       var testItem = id('qunit-test-output-' + details.testId);
       if (!tests || !testItem) {
@@ -7247,13 +7258,10 @@
       var good = details.passed;
       var bad = details.failed;
 
-      // This test passed if it has no unexpected failed assertions
-      var testPassed = details.failed > 0 ? details.todo : !details.todo;
       if (testPassed) {
         // Collapse the passing tests
         addClass(assertList, 'qunit-collapsed');
       } else {
-        stats.failedTests.push(details.testId);
         if (config.collapse) {
           if (!collapseNext) {
             // Skip collapsing the first failing test
@@ -7269,7 +7277,6 @@
       var testTitle = testItem.firstChild;
       var testCounts = bad ? "<b class='failed'>" + bad + '</b>, ' + "<b class='passed'>" + good + '</b>, ' : '';
       testTitle.innerHTML += " <b class='counts'>(" + testCounts + details.assertions.length + ')</b>';
-      stats.completed++;
       if (details.skipped) {
         testItem.className = 'skipped';
         var skipped = document.createElement('em');
